src/algebra/category/CommRing/limits.lean

```diff
@@ -95,7 +100,7 @@ An auxiliary declaration to speed up typechecking.
 -/
 def forget₂_AddCommMon_preserves_limits_aux (F : J ⥤ SemiRing) :
   is_limit ((forget₂ SemiRing AddCommMon).map_cone (limit_cone F)) :=
-AddCommMon.limit_cone_is_limit (F ⋙ forget₂ SemiRing AddCommMon)
+by convert AddCommMon.limit_cone_is_limit (F ⋙ forget₂ SemiRing AddCommMon)
```

Comment:
Isn't `by convert` dangerous for `defs` because it can introduce `eq.mpr`s?

Reply:
You're absolutely right that it introduces `eq.mpr`s. I don't know if it is harmful in this context, Another way is to do
```
begin
  let := AddCommMon.limit_cone_is_limit (F ⋙ forget₂ SemiRing AddCommMon),
  exact this
end
```
which introduced a `let` binding in the definition, but is also very quick. @semorrison , do you have an opinion on what is preferable in this kind of context?

Reply:
Yet another way is
```
begin
  have := AddCommMon.limit_cone_is_limit (F ⋙ forget₂ SemiRing AddCommMon),
  exact this
end
```
I thought that `have` was supposed to forget the data, but in fact it works fine, has the speedup, and the same `pp.all` output as the original term.

Reply:
I was going to suggest `have`. Remember that the `have` tactic does not produce a `have` term!

Reply:
I have switched to `by apply`, as advised on Zulip by Floris. It is essentially equivalent to the `have` solution above, but less verbose.

Reply:
Sounds great. Another possibility for dealing with "trivial" `eq.mpr`s is sometimes to wrap the tactic proof in `dsimp_result`.